typescript/packages/reality-data-client/src/RealityData.ts

```diff
@@ -218,9 +218,13 @@ export class ITwinRealityData implements RealityData {
 
         this._containerCache.setCache(newContainerCacheValue, access);
       }
-      return this._containerCache.getCache(access)!.url;
-    } catch (errorResponse: any) {
+      const cachedUrl = this._containerCache.getCache(access)?.url;
+      if (!cachedUrl) {
```

Comment:
I'm not sure this is right, if there's no url in the cache for the reality data, we request one from the API. At this point there should be a value in the cache or some error is thrown and caught. Perhaps just this is sufficient:
`return this._containerCache.getCache(access).url;`